src/overlays/actors/ovl_Bg_F40_Block/z_bg_f40_block.c

```diff
@@ -28,55 +38,340 @@ const ActorInit Bg_F40_Block_InitVars = {
     (ActorFunc)BgF40Block_Draw,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80BC4668[] = {
+static Vec3f D_80BC4620[] = {
+    { 1.0f, 0.0f, 0.0f },  { 0.0f, 1.0f, 0.0f },  { 0.0f, 0.0f, 1.0f },
+    { -1.0f, 0.0f, 0.0f }, { 0.0f, -1.0f, 0.0f }, { 0.0f, 0.0f, -1.0f },
+};
+
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 400, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 400, ICHAIN_CONTINUE),
     ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_STOP),
 };
 
-#endif
+s32 func_80BC3980(BgF40Block* this, GlobalContext* globalCtx) {
+    Vec3s* points;
+
+    this->unk_160 = 0;
+    this->unk_164 = 0;
+
+    if (BGF40BLOCK_GET_PATH(&this->dyna.actor) != 0x3F) {
```

Comment:
This check against 0x3F is repeated a few times, it may be useful to make a define for this value